crates/lambda-platform/src/winit/mod.rs

```diff
@@ -66,6 +78,77 @@ pub struct WindowHandle {
   pub monitor_handle: MonitorHandle,
 }
 
+// Should we take the loop as a field right here? Probably a ref or something? IDK
+pub struct WindowHandleBuilder {
+  window_handle: Option<Window>,
+  size: Option<WindowSize>,
+  monitor_handle: Option<MonitorHandle>,
+}
+
+impl WindowHandleBuilder {
+  /// Instantiate an empty builder
+  pub fn new() -> Self {
+    return Self;
+  }
+
+  /// Set the window size for the WindowHandle
+  fn with_window_size(
+    &mut self,
+    window_size: [u32; 2],
+    scale_factor: f64,
+  ) -> self {
```

Comment:
NIT: should use `Self` over `self` for the return signature

Reply:
Wouldn't I need to reconstruct the struct then? I figured it'd be more expensive than just modify the current instance so I went with this.

Reply:
Nevermind it's just the same.